src/components/FileTreeNode/index.tsx

```diff
@@ -101,13 +105,22 @@ class FileTreeNodeBase<TreeNodeType> extends React.Component<PublicProps> {
     let linterType = null;
     if (linterMessages && hasLinterMessages) {
       linterType = findMostSevereTypeForPath(linterMessages, node.id);
+      let title = getTitleForType(linterType, isFolder);
+
+      if (
+        linterType === 'notice' &&
+        linterMessages[node.id] &&
+        linterMessages[node.id].global.some(({ code }) =>
```

Comment:
I've considered this more and I think we should *only* mark files as known libraries if there is a *single* message (global or not) that has the matching code. For all other cases, let's highlight the file like we normally would.

The current patch could encourage the reviewer to ignore `FLAGGED_FILE_TYPE` (a `notice`) which might mean the file is binary, if I'm reading these docs right: https://github.com/mozilla/addons-linter/blob/5df2b4ecce7d9c9ad6abc0e6c7d1283a5b0e249c/docs/rules.md#package-layout (I could be wrong)

More importantly, the current patch seems like it would ignore any errors or warnings that are in the `byLine` mapping.

Reply:
> More importantly, the current patch seems like it would ignore any errors or warnings that are in the `byLine` mapping.

I don't think this is correct. the function checks for linter messages everywhere.

> I've considered this more and I think we should _only_ mark files as known libraries if there is a _single_ message (global or not) that has the matching code. For all other cases, let's highlight the file like we normally would.

well, if all messages are notice, it certainly means the messages are not super important. If `FLAGGED_*` is important, it should not be a `notice`. More importantly, if `KNOWN_LIBRARY` is less important than other notices, it should not have a `notice` level.

> Hiding known libraries has severe security implications so I want to be cautious with how we do it.

We are not hiding files.

Reply:
> I've considered this more and I think we should _only_ mark files as known libraries if there is a _single_ message (global or not) that has the matching code. For all other cases, let's highlight the file like we normally would.

I had this question too: https://github.com/mozilla/addons-code-manager/pull/418#discussion_r267805279

@kumar303 so, what should we do here in the end?

Reply:
> should we make sure that there is only/exactly one [known_library] message?

Yes, I think that is the safest way to do it. We should only consider it a known library file if there is exactly one message having the known library code across `global` and `byLine` mappings and any *future* mappings.

> We are not hiding files.

The purpose of dimming / graying the file is to let reviewers safely ignore the file so I'd say it is the equivalent of hiding the file from review.

>> More importantly, the current patch seems like it would ignore any errors or warnings that are in the byLine mapping.

> I don't think this is correct. the function checks for linter messages everywhere.

ok, good. Can you add an explicit test for that?

Reply:
> Yes, I think that is the safest way to do it.

Alrighty, let's do this! 💪